cmd/neofs-node/grpc.go

```diff
@@ -14,12 +15,53 @@ import (
 )
 
 func initGRPC(c *cfg) {
+	if c.cfgMorph.client == nil {
+		initMorphComponents(c)
+	}
+
+	// limit max size of single messages received by the gRPC servers up to max
+	// object size setting of the NeoFS network: this is needed to serve
+	// ObjectService.Replicate RPC transmitting the entire stored object in one
+	// message
```

Comment:
I think we'll have timeout problems with this setting for slow networks. It's perfectly fine for 50Gbit/s, but it will suck with 100 Mbit/s where 64M is more than 5 seconds. A single message will lock the line for this whole timeframe and no other messages could be transmitted. So I'd say we still need a stream (which can have a single message as we know), but then this setting can be optimized per-network.

Reply:
> no other messages could be transmitted

this is wrong: gRPC transmits messages from different RPCs concurrently, and each message is sent in small HTTP/2 DATA frames. The stream ensures a strict order of message transmission for this particular stream, rather than scheduling data frames sent over the network. Therefore, streams are not a way to transmit large data in gRPC - they are useful specifically for organizing messages (e.g. news feed, chats, blockchain events, etc.)

Reply:
Huh, I've not looked into HTTP/2 details of it. Then it should be OK, but at the same time there are some potential prioritization questions.

> open a socket dedicated only for replication

Seems to be a dead end to me, we better properly use the connection we have.

Reply:
> Seems to be a dead end to me

me too

> some potential prioritization questions

yeah it'd be interesting to design op prioritization model and test its impact to the system performance. Think it won't be hard to implement it on the gRPC lib basis